kep/260-network-aware-scheduling/README.md

```diff
@@ -0,0 +1,1500 @@
+	(...)
+	// 2) If all nodes were given the minimum score, return
+    (...)
+    // 3) Loop: for all scores 
+        (...) // If max != min
+            (...) // node_normalized_cost = MAX_SCORE * ( ( nodeScore - minCost) / (maxCost - minCost)
+            (...) // nodeScore = MAX_SCORE - node_normalized_cost
+        (...) // Else: If maxCost = minCost, avoid division by 0
+            (...) // node_normalized_cost = MAX_SCORE * - minCost
+            (...) // nodeScore = MAX_SCORE - node_normalized_cost
+
+    // 4) Return after the loop finishes 
+    return nil
+}
+```
+
+<!-- previous normalization:
+    // Lower scores correspond to lower latency
+   	// Get Min and Max Scores to normalize between framework.MaxNodeScore and framework.MinNodeScore
+   	minCost, maxCost := GetMinMaxScores(scores)
+   
+   	// If all nodes were given the minimum score, return
+   	if minCost == 0 && maxCost == 0 {
+   		return nil
+   	}
+   
+   	var normCost float64
+   	for i := range scores {
+   		if maxCost != minCost { // If max != min
+   			// node_normalized_cost = MAX_SCORE * ( ( nodeScore - minCost) / (maxCost - minCost)
+   			// nodeScore = MAX_SCORE - node_normalized_cost
+   			normCost = float64(framework.MaxNodeScore) * float64(scores[i].Score-minCost) / float64(maxCost-minCost)
+   			scores[i].Score = framework.MaxNodeScore - int64(normCost)
+   		} else { // If maxCost = minCost, avoid division by 0
+   			normCost = float64(scores[i].Score - minCost)
+   			scores[i].Score = framework.MaxNodeScore - int64(normCost)
+   		}
+   	}
+   	return nil
+-->
+
+We plan to combine our scoring plugin with other scoring plugins (e.g., `BalancedAllocation`, `LeastRequestedPriority`, etc). 
+We will attribute a higher weight to our plugin to prefer decisions focused on low latency. 
+For instance, consider the following scheduler config as an example:
+
+```yaml
+apiVersion: kubescheduler.config.k8s.io/v1beta1
+kind: KubeSchedulerConfiguration
+leaderElection:
+  leaderElect: false
+clientConnection:
+  kubeconfig: "REPLACE_ME_WITH_KUBE_CONFIG_PATH"
+profiles:
+- schedulerName: network-aware-scheduler
+  plugins:
+    queueSort:
+      enabled:
+      - name: TopologicalSort
+      disabled:
+      - name: "*"
+    filter:
+      enabled:
+      - name: NetworkOverhead
+    score:
+      disabled: # Preferably avoid the combination of these two plugins with the NetworkOverhead
+      - name: MostAllocated
+      - name: LeastAllocated
+      enabled: # A higher weight is given to NetworkOverhead to favor allocation schemes with lower latency.
+      - name: NetworkOverhead
+        weight: 5
+      - name: BalancedAllocation
+        weight: 1
+  pluginConfig:
+  - name: TopologicalSort
+    args:
+      kubeConfigPath: "/etc/kubernetes/scheduler.conf"
+      namespaces:
+      - "default"
+  - name: NetworkOverhead
+    args:
+      kubeConfigPath: "/etc/kubernetes/scheduler.conf"
```

Comment:
now it's not needed.

Reply:
Thanks @Huang-Wei. I have updated the scheduler config for v1beta2:

```yaml
apiVersion: kubescheduler.config.k8s.io/v1beta2
kind: KubeSchedulerConfiguration
leaderElection:
  leaderElect: false
clientConnection:
  kubeconfig: "REPLACE_ME_WITH_KUBE_CONFIG_PATH"
profiles:
- schedulerName: network-aware-scheduler
  plugins:
    queueSort:
      enabled:
      - name: TopologicalSort
      disabled:
      - name: "*"
    filter:
      enabled:
      - name: NetworkOverhead
    score:
      disabled: # Preferably avoid the combination of NodeResourcesFit with NetworkOverhead
      - name: NodeResourcesFit
      enabled: # A higher weight is given to NetworkOverhead to favor allocation schemes with lower latency.
      - name: NetworkOverhead
        weight: 5
      - name: BalancedAllocation
        weight: 1
  pluginConfig:
  - name: TopologicalSort
    args:
      namespaces:
      - "default"
  - name: NetworkOverhead
    args:
      namespaces:
      - "default"
      weightsName: "UserDefined" # weights applied by the plugin
      networkTopologyName: "net-topology-test" # networkTopology CR used by the plugin
```